### PR TITLE
[RFC] common/environment/setup/install.sh: always add log service

### PR DESCRIPTION
**Pull request by classabbyamp**
- common/environment/setup/install.sh: always add log service
    - overridable by having a log service at `$pkgname/files/$service/log`
    - uses a sane default run script with the service name set as tag and daemon facility
    - warns if stderr is not redirected in the main service
    - example of warnings:
```
=> soju-0.5.2_1: running post_install ...
=> WARNING: soju-0.5.2_1: vsv: service 'soju' does not contain 'exec 2>&1' to log stderr
=> WARNING: soju-0.5.2_1: vsv: overriding default log service
```
- Manual.md: document vsv log autocreation
- *: remove log dir en-masse

#### Testing the changes
- I tested the changes in this PR: **YES**

### TODO
- [x] go through all removed log services and see if any need overrides
    - candidates at a glance: cronie, dcron, dkimproxy, dnscrypt-proxy, dq, postgres13/14, sqmail
- [x] add/fix redirection to main run file of all services

### Questions
- ~~Should there be a possible second vlogger argument that sets the facility? This would remove the need for overrides for the few services that set cron/mail as facility~~
    - added this for ease-of-use

[ci skip]

### CODE REVIEW

**Review by the-maldridge — APPROVED** [review]
Approved for general concept.  It would be nice to do a bulk change on existing service files to make sure redirection is setup correctly.

**Comment by classabbyamp** [comment]
yes, I'll add that to the todo

File: common/environment/setup/install.sh [hunk 1]
```diff
@@ -34,17 +34,22 @@ _vsv() {
 	vmkdir etc/sv
 	vcopy "${FILESDIR}/$service" etc/sv
 	if [ ! -L $svdir/run ]; then
+		grep -q 'exec 2>&1' $svdir/run || msg_warn "$pkgver: vsv: service '$service' does not contain 'exec 2>&1' to log stderr\n"
```
Comment (paper42) [inline common/environment/setup/install.sh:37]: `&` is a special character, but I am not sure if this matters
```suggestion
		grep -F -q 'exec 2>&1' $svdir/run || msg_warn "$pkgver: vsv: service '$service' does not contain 'exec 2>&1' to log stderr\n"
```

File: common/environment/setup/install.sh [hunk 2]
```diff
@@ -34,17 +34,22 @@ _vsv() {
...
-	if [ -d $svdir/log ]; then
-		ln ${LN_OPTS} /run/runit/supervise.${service}-log $svdir/log/supervise
-		if [ -e $svdir/log/run ] && [ ! -L $svdir/log/run ]; then
-			chmod 755 ${PKGDESTDIR}/etc/sv/${service}/log/run
-		fi
+	if [ -d $svdir/log ] || [ -L $svdir/log ]; then
+		msg_warn "$pkgver: vsv: overriding default log service\n"
+	else
+		mkdir $svdir/log
+		printf '#!/bin/sh\nexec vlogger -p daemon -t %s\n' "$service" > $svdir/log/run
```
Comment (sgn) [inline common/environment/setup/install.sh:48]:
```suggestion
		cat <<-EOF  >$svdir/log/run
		#!/bin/sh
		exec vlogger -p daemon -t $service
		EOF
```

While the printf is correct, it isn't visualized as good as heredoc.

**Comment by classabbyamp** [comment]
remaining question: why would a service need to chpst for svlogd? is this necessary with vlogger? see [dnscrypt-proxy](https://github.com/void-linux/void-packages/blob/master/srcpkgs/dnscrypt-proxy/files/dnscrypt-proxy/log/run) and [dqcache](https://github.com/void-linux/void-packages/blob/master/srcpkgs/dq/files/dqcache/log/run) services

**Comment by ahesford** [comment]
A service shouldn't need `chpst` with `vlogger`, but those logs are using `svlogd` to build dedicated log directories and want to keep them owned by the dedicated user.

**Comment by classabbyamp** [comment]
should they be migrated to the standard vlogger setup?

**Pull request merged**